osu.Game/Screens/Play/GameplayBeatmap.cs

```diff
@@ -38,5 +41,14 @@ public BeatmapInfo BeatmapInfo
         public IEnumerable<BeatmapStatistic> GetStatistics() => PlayableBeatmap.GetStatistics();
 
         public IBeatmap Clone() => PlayableBeatmap.Clone();
+
+        public IBindable<bool> Passing => passing;
+
+        private readonly BindableBool passing = new BindableBool(true);
+
+        public void OnNewResult(JudgementResult result)
+        {
+            passing.Value = result.Type > HitResult.Miss;
+        }
```

Comment:
Perhaps a little monothematic of me, but in light of #8857 I'm wondering what direction this should go towards. The name of `Passing` could be a little misleading (could be misinterpreted for checking health), but the bigger issue I'm seeing is that the taiko pippidon mascot appears to need a *lot* more information to reimplement stable logic, namely:

* if the last result was a hit or not
* whether the last judgement affects combo (only combo-affecting judgements change state)
* what was the total combo at the judgement (special animation every 50 combo)
* what the judgement type was (special animation on every swell passed)
* once the map is done, whether the play is a pass or a fail (currently don't see an easy way to access health info from a skinnable drawable)

To get most of that info currently I'm jamming the entire `JudgementResult` into the mascot, but I'm a little worried that we might soon have two ways of doing essentially the same thing. Jamming all of the above data into `GameplayBeatmap` seems like an... inelegant out, but storing the last result in the beatmap might also not be great.

Reply:
exposing a bindable `LastJudgement` in `GameplayBeatmap` might be enough to cover all of that?

Reply:
Health is the possible one exception (taiko is weird in that `FailedAtJudgement` doesn't really work, since the fail doesn't happen until everything is actually judged) but as for everything else, seems like a possible way forward, sure.

Reply:
I've updated to this. I guess the only thing remaining is combo and health.

Reply:
Cool 👍 Once this is in I'll merge in these changes to that mascot PR and reuse. I'd say the health stuff can come as a separate PR as the changes I have in there are extensive enough as-is.